### ISSUE TITLE
Error on @ConditionalOnExpression

### ISSUE THREAD

**Issue by hugograf** [spring-projects/sts4#1349]
**Describe the bug**
There is an error for @ConditionalOnExpression(“${gt.use.websocket:false}”).

**To Reproduce**
Annotation
```
@ConditionalOnExpression("${gt.use.websocket:false}")
```
produces
```
SPEL: mismatched input 'gt' expecting {'++', '+', '--', '-', '(', '#', '@', '^[', '![', '!', '&', '?[', '$[', LCURLY, 'true', 'false', 'new', 'null', 'T', IDENTIFIER, NUMERIC_LITERAL, STRING_LITERAL}
```

**Version**
```
Spring Tool Suite 4 

Version: 4.24.0.RELEASE
Build Id: 202407191504
```

**Comment by martinlippert** [comment]
Thanks a lot @hugograf for reporting. Moved this to the 4.26.0 release, will unfortunately not make it at the last minute into the upcoming 4.25.0 release.

**Comment by BoykoAlex** [comment]
Fixed with fca3fc0566a202b8ed772cb1c6709f60a7bbf21e

**Comment by martinlippert** [comment]
@hugograf Thanks again for reporting. If you want to try the fix, feel free to grab a snapshot build from https://cdn.spring.io/spring-tools/snapshot/STS4/nightly-distributions.html
